owlkettle/mainloop.nim

```diff
@@ -65,11 +65,28 @@ proc loadStylesheet*(path: string, priority: int = DEFAULT_PRIORITY): Stylesheet
     raise newException(IOError, $error[].message)
   result = Stylesheet(provider: provider, priority: priority)
 
-type AppConfig* = object of RootObj
-  widget*: Widget
-  icons*: seq[string]
-  darkTheme*: bool
-  stylesheets*: seq[Stylesheet]
+type 
+  AppConfig* = object of RootObj
+    widget*: Widget
+    icons*: seq[string]
+    darkTheme*: bool
+    stylesheets*: seq[Stylesheet]
+  
+  AppContext* = object
+    config*: AppConfig
```

Comment:
To fix the comment above this would have to be generic over different `AppConfig` types.

Reply:
This is weird. It also fails for ref objects:

```nim
type
  A = ref object of RootObj
  B = ref object of A
  
  C[T: A] = object
    x: T

discard C[B](x: B())
```

However when the same pattern is used for the generic instantiation of a proc, it works:

```nim
proc test[T: A](): T =
  result = T()

echo test[A]().repr
echo test[B]().repr
```

Possibly a compiler issue, but even if not, this is inconsistent behaviour.